src/Sylius/Behat/spec/Context/Domain/PromotionContextSpec.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Behat\Context\Domain;
+
+use Behat\Behat\Context\Context;
+use Doctrine\DBAL\Exception\ForeignKeyConstraintViolationException;
+use PhpSpec\Exception\Example\FailureException;
+use PhpSpec\Exception\Example\NotEqualException;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Behat\Context\Domain\PromotionContext;
+use Sylius\Component\Core\Repository\PromotionRepositoryInterface;
+use Sylius\Component\Core\Test\Services\SharedStorageInterface;
+use Sylius\Component\Promotion\Model\CouponInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+/**
+ * @mixin PromotionContext
+ *
+ * @author Jan Góralski <jan.goralski@lakion.com>
+ */
+class PromotionContextSpec extends ObjectBehavior
+{
+    function let(
+        SharedStorageInterface $sharedStorage,
+        PromotionRepositoryInterface $promotionRepository,
+        RepositoryInterface $couponRepository
+    ) {
+        $this->beConstructedWith($sharedStorage, $promotionRepository, $couponRepository);
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('Sylius\Behat\Context\Domain\PromotionContext');
+    }
+
+    function it_implements_context_interface()
+    {
+        $this->shouldImplement(Context::class);
+    }
+
+    function it_removes_a_coupon(
+        RepositoryInterface $couponRepository,
+        SharedStorageInterface $sharedStorage,
+        CouponInterface $coupon
+    ) {
+        $coupon->getId()->willReturn(5);
+        $sharedStorage->set('coupon_id', 5)->shouldBeCalled();
+        $couponRepository->remove($coupon)->shouldBeCalled();
+
+        $this->iDeleteCoupon($coupon);
+    }
+
+    function it_throws_exception_when_there_was_a_problem_with_removing_a_coupon(
+        RepositoryInterface $couponRepository,
+        SharedStorageInterface $sharedStorage,
+        CouponInterface $coupon
```

Comment:
Can we put these arguments (and in other methods) in the same order, as in constructor? :smile: #structuralNazi